src/net/zephyr/network.c

```diff
@@ -0,0 +1,389 @@
+{
+    _zn_socket_result_t r;
+    r.tag = _z_res_t_OK;
+
+    _Z_DEBUG_VA("Binding UDP Socket to: %s:%d\n", addr, port);
+    struct sockaddr_in saddr;
+
+    r.value.socket = socket(PF_INET, SOCK_DGRAM, 0);
+
+    if (r.value.socket < 0)
+    {
+        r.tag = _z_res_t_ERR;
+        r.value.error = r.value.socket;
+        r.value.socket = 0;
+        return r;
+    }
+
+    memset(&saddr, 0, sizeof(saddr));
+    saddr.sin_family = AF_INET;
+    saddr.sin_port = htons(port);
+
+    if (inet_pton(AF_INET, addr, &saddr.sin_addr) <= 0)
+    {
+        r.tag = _z_res_t_ERR;
+        r.value.error = _zn_err_t_INVALID_LOCATOR;
+        r.value.socket = 0;
+        return r;
+    }
+
+    if (bind(r.value.socket, (struct sockaddr *)&saddr, sizeof(saddr)) < 0)
+    {
+        r.tag = _z_res_t_ERR;
+        r.value.error = _zn_err_t_INVALID_LOCATOR;
+        r.value.socket = 0;
+        return r;
+    }
+
+    struct timeval timeout;
+    timeout.tv_sec = 0;
+    timeout.tv_usec = timeout_usec;
+    if (setsockopt(r.value.socket, SOL_SOCKET, SO_RCVTIMEO, (void *)&timeout, sizeof(struct timeval)) == -1)
+    {
+        r.tag = _z_res_t_ERR;
+        r.value.error = errno;
+        close(r.value.socket);
+        r.value.socket = 0;
+        return r;
+    }
+
+    // NOTE(esteve): SO_SNDTIMEO not supported in Zephyr
+/*
+    if (setsockopt(r.value.socket, SOL_SOCKET, SO_SNDTIMEO, (void *)&timeout, sizeof(struct timeval)) == -1)
+    {
+        r.tag = _z_res_t_ERR;
+        r.value.error = errno;
+        close(r.value.socket);
+        r.value.socket = 0;
+        return r;
+    }
+*/
+    return r;
+}
+
+_zn_socket_result_t _zn_open_tx_session(const char *locator)
+{
+    _zn_socket_result_t r;
+    r.tag = _z_res_t_OK;
+    char *l = strdup(locator);
+    _Z_DEBUG_VA("Connecting to: %s:\n", locator);
+    char *tx = strtok(l, "/");
+    if (strcmp(tx, "tcp") != 0)
+    {
+        fprintf(stderr, "Locator provided is not for TCP\n");
+        exit(1);
+    }
+    char *addr_name = strdup(strtok(NULL, ":"));
+    char *s_port = strtok(NULL, ":");
+
+    int status;
+    char ip_addr[INET6_ADDRSTRLEN];
+    struct sockaddr_in *remote;
+    struct addrinfo *res;
+    status = getaddrinfo(addr_name, s_port, NULL, &res);
+    free(addr_name);
+    if (status == 0 && res != NULL)
+    {
+        void *addr;
+        remote = (struct sockaddr_in *)res->ai_addr;
+        addr = &(remote->sin_addr);
+        inet_ntop(res->ai_family, addr, ip_addr, sizeof(ip_addr));
+    }
+    freeaddrinfo(res);
+
+    int port;
+    sscanf(s_port, "%d", &port);
+
+    _Z_DEBUG_VA("Connecting to: %s:%d\n", addr_name, port);
+    free(l);
+    struct sockaddr_in serv_addr;
+
+    r.value.socket = socket(PF_INET, SOCK_STREAM, 0);
+
+    if (r.value.socket < 0)
+    {
+        r.tag = _z_res_t_ERR;
+        r.value.error = r.value.socket;
+        r.value.socket = 0;
+        return r;
+    }
+
+    // NOTE(esteve): SO_KEEPALIVE and SO_LINGER not supported in Zephyr
```

Comment:
I would simply remove the commented code and leave the note saying that those flags are not supported in Zephyr.